cmd/mattermost/commands/user.go

```diff
@@ -339,10 +339,21 @@ func userCreateCmdF(command *cobra.Command, args []string) error {
 		Locale:    locale,
 	}
 
-	if ruser, err := a.CreateUser(user); err != nil {
+	ruser, err := a.CreateUser(user)
+	if ruser == nil {
 		return errors.New("Unable to create user. Error: " + err.Error())
-	} else if systemAdmin {
-		a.UpdateUserRoles(ruser.Id, "system_user system_admin", false)
+	}
+
+	if systemAdmin {
+		if _, err := a.UpdateUserRoles(ruser.Id, "system_user system_admin", false); err != nil {
+			return errors.New("Unable to make user system admin. Error: " + err.Error())
+		}
+	} else {
+		// This else case exists to prevent the first user created from being
+		// created as a system admin unless explicity specified.
+		if _, err := a.UpdateUserRoles(ruser.Id, "system_user", false); err != nil {
+			return errors.New("Unable to prevent user from being system admin if this is the first user. Error: " + err.Error())
```

Comment:
Does that error message make sense? Seems to contradict what the comment is saying

Reply:
Does it? I think it says the same thing.

Reply:
They way I interpret it is that the comment is saying this line of code will prevent the first user being a system admin. The error seems to be saying that it couldn't be prevented because the first user must be a system admin

Reply:
Ya. That's right. The code is trying to prevent the first user from being a system admin. So if it can't do that it is "unable to prevent user from being system admin".

Reply:
Hopfully made it more clear?